### ISSUE TITLE
Treemap tile custom click handler

### ISSUE THREAD

**Issue by francescocretti** [d3plus/d3plus-react#54]
Maybe I'm missing something here, but I cannot find in any place in the documentation the possibility to add a custom click handler on the Treemap tiles.

What I want to do is really simple: open a link when the user clicks a tile on the treemap. Is not really possible to do that?

Thanks a lot in advance,

Francesco

**Comment by davelandry** [comment]
@francescocretti you are right, we don't have any examples that show this! I will repurpose this issue to remind us to create an example, but in the meanwhile:

The `on` method of all visualizations allows you to map event functions to the shapes drawn by d3plus. For example:

```jsx
config={{
  ...,
  on: {
    click: (d, i) => {
      console.log("clicked data point:", d);
    }
  },
  ...
}}
```

Furthermore, these events can be namespaced to specifically target either the `shapes` drawn in the visualization itself, or the shapes drawn by the `legend`:

```jsx
config={{
  ...,
  on: {
    "click.shape": (d, i) => {
      console.log("I'm a Treemap square!");
    },
    "click.legend": (d, i) => {
      console.log("I'm a Legend square!");
    }
  },
  ...
}}
```
